Update TCK for EL 5.0

markt-asf opened the pull request:
---
name: Pull Request
about: Create a pull request for a Platform TCK change
title: 'Update TCK for EL 5.0'
labels: ''
assignees: ''

---
**Fixes Issue**
#749 

**Related Issue(s)**
#740

**Describe the change**
Updates existing tests and adds new tests for EL 5.0

**Additional context**
This PR is currently in draft as it is dependent on a number of changes before it can be applied
- The TCK needs to be compiled with Java 11 and with Java 11 as the minimum version (#740)
- Glassfish needs to update to at least EL 5.0.0-M1 to pick up the API changes under test
- The tests added for MethodReference need to be confirmed as valid. I think they are correct but I can't check them until the previous two dependencies have been met

Additionally, this is by far the largest change I have made to a TCK. I suspect I may not have followed the expected approach in some cases and some refactoring will be required.

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @scottmarlow

Thihup commented:
@markt-asf Would it be possible to add some TCK tests related to Map literal? I think this part of the spec, the TCK doesn't cover it very well.

https://github.com/eclipse-ee4j/el-ri/issues/161

gurunrao reviewed (commented):
update copyright to 2021.

markt-asf commented:
Thanks for the JDK 11 updates. I am able to build the EL TCK against a local build of the EL API now.

It appears I have a couple of issues to fix in the original draft as well as the comments above to address. I'll push an updated version of the PR when I have completed that work.

markt-asf commented on src/com/sun/ts/tests/el/api/jakarta_el/methodexpression/ELClient.java:
```diff
@@ -419,4 +422,227 @@ public void methodExpressionSerializableTest() throws Fault {
+        pass = false;
+      }
+    } catch (Exception ex) {
+      throw new Fault(ex);
```
Done.

markt-asf commented on src/com/sun/ts/tests/el/api/jakarta_el/methodexpression/ELClient.java:
```diff
@@ -419,4 +422,227 @@ public void methodExpressionSerializableTest() throws Fault {
+
+  /*
+   * @testName: methodExpressionMatchingExactPreferredTest
+   * 
```
Done.

markt-asf commented:
This updated PR addresses all the review comments and includes the requests additional TCK tests for Map Literals. It also includes similar tests for Set and List literals. Thanks for the suggestion - I found and fixed a bug in Tomcat's EL implementation as a result.
I'm leaving this PR as draft as there is one final EL issue to resolve. I hope to get to that today and once the API changes are complete I'll wrap up this PR and move it out of draft.

markt-asf commented:
This PR now contains all the necessary changes for the changes in EL 5.0

markt-asf commented:
Just a rebase. No other changes.

markt-asf commented:
This should now be complete for EL 5.0

scottmarlow commented on src/com/sun/ts/tests/common/el/api/expression/ExpressionTest.java:
```diff
@@ -87,6 +89,85 @@ public static boolean testMethodInfo(MethodInfo minfo, String expectedName,
+    Annotation[] annotations = mref.getAnnotations();
+    Object[] parameterValues = mref.getEvaluatedParameters();
+    
+    if (base == null) {
```
@gurunrao is this change blocking your approval or are there other pending changes your waiting on to approve?

gurunrao replied:
> @gurunrao is this change blocking your approval or are there other pending changes your waiting on to approve?

ya, was waiting for that. approved anyway.

gurunrao commented:
> This should now be complete for EL 5.0

@markt-asf - Please resolve conflicts at `"src/com/sun/ts/tests/signaturetest/signature-repository/jakarta.el.sig_4.0_se8"` for the merge to proceed.

markt-asf commented:
Force push should have resolved the conflicts.
